src/overflow.ts

```diff
@@ -0,0 +1,73 @@
+import { AttributePart, noChange } from 'lit-html';
+import { Directive, directive, DirectiveResult } from 'lit-html/directive.js';
+
+type OnOverflow = (
+	visibleElements: Set<HTMLElement>,
+	overflownElements: Set<HTMLElement>,
+) => void;
+
+function isEntryHidden(el: IntersectionObserverEntry) {
+	return el.boundingClientRect.height === 0;
+}
+
+class OverflowDirective extends Directive {
```

Comment:
You should extend `AsyncDirective` and implement `disconnected` and `reconnected`, so that you can properly clean up. Failing to clean up the observer and event handler will hold onto the DOM of the buttons and prevent garbage collection.